### ISSUE TITLE
An advice for set_params() when using kvstore

### ISSUE THREAD

**Issue by starimpact** [apache/incubator-mxnet#6005]
finally, I found why my training loss can not decreased with kvstore.
that is because kvstore can not get the new weight value after its init.
kvstore only updates the weight in itself.
set_param() can not affect the weight of kvstore actually!
but, In my training, I need to exchange the weight value periodically.
So, I suggest kvstore can accept the new weight value by using set_params() in the training time.

@piiswrong @tqchen

**Comment by starimpact** [comment]
Can I delte the 
```cpp
      CHECK(local_.find(keys[i]) == local_.end())
          << "duplicate init of key " << keys[i];
```
of 
```cpp
  void Init(const std::vector<int>& keys,
            const std::vector<NDArray>& values) override {
    for (size_t i = 0; i < keys.size(); ++i) {
      CHECK(local_.find(keys[i]) == local_.end())
          << "duplicate init of key " << keys[i];
      local_[keys[i]] = values[i].Copy(pinned_ctx_);
      comm_->Init(keys[i], values[i].shape());
    }
  }
```
because, I want to reinitialize the weight value on training procedure...

@piiswrong

**Comment by starimpact** [comment]
I add Reset function into kvstore and related files, do you need it? I can pull a request.

**Comment by starimpact** [comment]
@piiswrong

**Comment by starimpact** [comment]
I'm maintaining a version based on mxnet_v0.8.0:
https://github.com/starimpact/mxnet_v0.8.0

**Comment by starimpact** [comment]
I just did a pull request.

**Comment by starimpact** [comment]
the problem wanted to solve:
assume there is a very big weight parameter weight_Z, like 100Wx128 dims. we can only update a part of it each time, such as 4Wx128.
I have implemented in the local calculation. but It seems to change a lot code if I want to do it in a distribution calculation.It is feasible, but just a little not easy to do it.

**Comment by eric-haibin-lin** [comment]
Which layer only has 4Wx128 gradients?

**Comment by starimpact** [comment]
the last layer, such as softmax or proxy.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
